ZipBuilder/Sources/ZipBuilder/FirebaseBuilder.swift

```diff
@@ -0,0 +1,157 @@
+struct FirebaseBuilder {
+  /// ZipBuilder instance.
+  private let zipBuilder: ZipBuilder
+
+  /// Default initializer. If allSDKsPath and currentReleasePath are provided, it will also verify
+  /// that the
+  ///
+  /// - Parameters:
+  ///   - paths: Paths that are needed throughout the process of packaging the Zip file.
+  ///   - customSpecRepo: A custom spec repo to be used for fetching CocoaPods from.
+  init(zipBuilder: ZipBuilder) {
+    self.zipBuilder = zipBuilder
+  }
+
+  func build(in projectDir: URL) {
+    // Build the zip file and get the path.
+    do {
+      let artifacts = try builder.buildAndAssembleFirebaseRelease(inProjectDir: projectDir)
+      let firebaseVersion = artifacts.firebaseVersion
+      let location = artifacts.outputDir
+      print("Firebase \(firebaseVersion) directory is ready to be packaged: \(location)")
+
+      // Package carthage if it's enabled.
+      var carthageRoot: URL?
+      if let carthageJSONDir = args.carthageDir {
+        do {
+          print("Creating Carthage release...")
+          // Create a copy of the release directory since we'll be modifying it.
+          let carthagePath =
+            location.deletingLastPathComponent().appendingPathComponent("carthage_build")
+          let fileManager = FileManager.default
+          fileManager.removeIfExists(at: carthagePath)
+          try fileManager.copyItem(at: location, to: carthagePath)
+
+          // Package the Carthage distribution with the current directory structure.
+          let carthageDir = location.deletingLastPathComponent().appendingPathComponent("carthage")
+          fileManager.removeIfExists(at: carthageDir)
+          var output = carthageDir.appendingPathComponent(firebaseVersion)
+          if let rcNumber = args.rcNumber {
+            output.appendPathComponent("rc\(rcNumber)")
+          } else {
+            output.appendPathComponent("latest-non-rc")
+          }
+          try fileManager.createDirectory(at: output, withIntermediateDirectories: true)
+          CarthageUtils.generateCarthageRelease(fromPackagedDir: carthagePath,
+                                                templateDir: args.templateDir,
+                                                jsonDir: carthageJSONDir,
+                                                firebaseVersion: firebaseVersion,
+                                                coreDiagnosticsPath: artifacts.carthageDiagnostics,
+                                                outputDir: output)
+
+          // Remove the duplicated Carthage build directory.
+          fileManager.removeIfExists(at: carthagePath)
+          print("Done creating Carthage release! Files written to \(output)")
+
+          // Save the directory for later copying.
+          carthageRoot = carthageDir
+        } catch {
+          // TODO: This can fail on CI due to size requirements, let's fail gracefully in the meantime
```

Comment:
Was this comment from before? The issue was resolved on Kokoro with the `-outputDir` fix so it can probably be re-enabled.

Reply:
Yes. This is unchanged. I reenabled.